### PR TITLE
Test Engineering strategy

### PR DESCRIPTION
**Pull request by rbillings**
Created draft of information related to Normandy Test Engineering. Comments welcome - including opinions on content, placement of doc [should it live in the QA folder?], etc.

### CODE REVIEW

File: docs/qa/test_engineering [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Normandy Test Engineering
```
Comment (mythmon) [inline docs/qa/test_engineering]: You'll need to name this file `test_engineering.rst`.

File: docs/qa/test_engineering [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+This is the repository for all information related to test engineering for the Normandy project.  This will outline our resources and test process for releases.
+
+The primary focus of testing will be on the Firefox client and the API server. The Normandy Firefox client follows the release process and train schedule for the Firefox browser. The Normandy server is tested using the API server and recipes.
+
+The Delivery Console is not currently being tested, as it’s a much lower priority. 
+
+The test strategy incorporates manual and automated testing. Different approaches are outlined for each area. 
+
+Normandy Server
+--------------
```
Comment (mythmon) [inline docs/qa/test_engineering]: RST is picky and requires that the underline here is the exact same length as the line it is underlining.

File: docs/qa/test_engineering [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Automation:
+Test Engineer: b4hand
+Manual tests: none
+Status: In process
+
+Normandy Firefox client
+-----------------------
+     The Normandy Firefox client queries the Normandy server for instructions and picks up recipe changes. 
+QA: cmuresan
+Automation: https://github.com/18epedersen/Normandy-e2e-tests Status: outdated
```
Comment (mythmon) [inline docs/qa/test_engineering]: This isn't correct. That test suite was for the predecessor to Delivery Console, not for the Normandy Firefox client.

File: docs/qa/test_engineering [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+The test strategy incorporates manual and automated testing. Different approaches are outlined for each area. 
+
+Normandy Server
+--------------
+The Normandy Server is an API server with recipes. This feature is high priority, and needs to maintain a high level of service. There are two versions with different functionality. V1 is the server clients connect to. V3 is the server that Admins connect to.
+
+- v1
+Automation:
+Test Engineer: chartjes
+Status:  In process
```
Comment (mythmon) [inline docs/qa/test_engineering]: I suspect the sections like this will probably need to be reworked to be valid reStrucuturedText document. Sphinx has docs about RST here: http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html

File: docs/qa/test_engineering (hunk 1 above)
Comment (rbillings) [reply docs/qa/test_engineering]: done

File: docs/qa/test_engineering (hunk 2 above)
Comment (rbillings) [reply docs/qa/test_engineering]: fixed

File: docs/qa/test_engineering (hunk 4 above)
Comment (rbillings) [reply docs/qa/test_engineering]: I reviewed that page and am not sure what needs to be done for formatting as it was intended to be just a list of info.

Comment (mythmon) [reply docs/qa/test_engineering]: You can see [in this Github preview of the document](https://github.com/mozilla/normandy/pull/1793/files?short_path=99599b2#diff-99599b2d5a06775a6b1849ab92de1c6a) that it is currently rendering everything on a single line, with the "v1" and "v3" as interspersed bullet points. That's not very readable.

**Comment by mythmon** [comment]
The CI run to build the docs [failed](https://circleci.com/gh/mozilla/normandy/8316?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) due to bad RST syntax. Unfortunately, RST is a bit pickier than Markdown. If you're going to be writing docs for Normandy, I suggest building them locally and seeing how they look when rendered. We have some [short instructions](https://circleci.com/gh/mozilla/normandy/8316?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) on how to build the documentation in a local checkout as a part of our documentation.

You can learn more about RST and Sphinx (our documentation tool) [on their website](http://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html).

**Comment by peterbe** [comment]
Sorry for butting in but a long time I ago I wrote this: https://gist.github.com/peterbe/73b8da9262a2ce1dba2edb4ea2daba66 to build the normandy docs locally. I don't even know if it still works but it made it possible for me to run it quickly.

**Comment by rbillings** [comment]
The only thing not addressed is adding a link to this doc from the index - I will do that if it passes inspection.

File: docs/qa/test_engineering.rst [hunk 5]
```diff
@@ -9,36 +9,39 @@ The Delivery Console is not currently being tested, as it’s a much lower prior
...
-Automation: 
-Test Engineer: b4hand
-Manual tests: https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918
-Status: b4hand is reviewing and fixing outdated UI Automation tests. Ciprian is reviewing and updating manual tests.
+The Normandy Firefox client queries the Normandy server for instructions and picks up recipe changes. 
+
+* QA: cmuresan
+* Automation: 
+* Test Engineer: b4hand
+* [Manual tests](https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918)
```
Comment (mythmon) [inline docs/qa/test_engineering.rst]: This isn't how RST handles links. Here's an example of how to do an inline link:

https://github.com/mozilla/normandy/blob/003cdfa7621c74630432a0b7134ab818f8ddde74/docs/dev/remote-settings.rst#L56-L58

And here is an example of a not-inline link, where the URL is placed separately to improve code readability.

https://github.com/mozilla/normandy/blob/003cdfa7621c74630432a0b7134ab818f8ddde74/docs/dev/install.rst#L10-L20

File: docs/qa/test_engineering.rst [hunk 6]
```diff
@@ -9,36 +9,39 @@ The Delivery Console is not currently being tested, as it’s a much lower prior
 The test strategy incorporates manual and automated testing. Different approaches are outlined for each area. 
 
 Normandy Server
----------------
+--------------
```
Comment (mythmon) [inline docs/qa/test_engineering.rst]: RST warns that this udnerline is too short, compared to the text it is underlining. Our set up treats warnings as errors, so this is a problem.

File: docs/qa/index.rst [hunk 7]
```diff
@@ -3,7 +3,8 @@ QA Documentation
 These documents are aimed at QA testers working on Normandy.
 
 .. toctree::
-   :maxdepth: 2
+   :maxdepth: 3
```
Comment (mythmon) [inline docs/qa/index.rst]: Why did you change this to 3?

File: docs/qa/test_engineering.rst [hunk 8]
```diff
@@ -30,19 +30,19 @@ The Normandy Firefox client queries the Normandy server for instructions and pic
 * QA: cmuresan
 * Automation: 
 * Test Engineer: b4hand
-* [Manual tests](https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918)
+* _Manual tests: https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918
```
Comment (mythmon) [inline docs/qa/test_engineering.rst]: This doesn't work because there is no link between "Manual" and "tests". It's looking for a link with an ID of `Manual`.

The correct version would be either this inline version:

```rst
* `Manual tests <https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918>` are maintained on Test Rail
```

OR this not-inline version:

```rst
* `Manual tests`_ are maintained on Test Rail

.. _Manual tests: https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918
```

File: docs/qa/test_engineering.rst [hunk 9]
```diff
@@ -30,19 +30,19 @@ The Normandy Firefox client queries the Normandy server for instructions and pic
...
 * **Status:** b4hand is reviewing and fixing outdated UI Automation tests. Ciprian is reviewing and updating manual tests.
 
 Normandy Delivery Console
 -------------------------
 This is a tool to edit API recipes by Admins. It’s also referred to as the Admin panel. It is a single page application that is only used by internal users. If there is a problem it is reported by internal users, and is considered an inconvenience, not a high priority.
 
 * QA/Test Engineer: none
-* [Automation](https://github.com/18epedersen/Normandy-e2e-tests)
+* _Automation: https://github.com/18epedersen/Normandy-e2e-tests
```
Comment (mythmon) [inline docs/qa/test_engineering.rst]: This link needs fixed.

File: docs/qa/test_engineering.rst [hunk 10]
```diff
@@ -30,19 +30,19 @@ The Normandy Firefox client queries the Normandy server for instructions and pic
...
 	* **Status:** needs updating
-* [Manual tests](https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918)
+* _Manual tests: https://testrail.stage.mozaws.net/index.php?/suites/view/232&group_by=cases:section_id&group_order=asc&group_id=16918
```
Comment (mythmon) [inline docs/qa/test_engineering.rst]: This link needs fixed.

**Comment by mythmon** [comment]
Oops, this time it's my fault (I didn't actually check my syntax before). The inline form needs a trailing backslash to make it a link. I'll fix that up.
